warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ShortCircuitBaseVisitor.java

```diff
@@ -0,0 +1,216 @@
+import org.apache.commons.jexl2.parser.ASTFalseNode;
+import org.apache.commons.jexl2.parser.ASTFloatLiteral;
+import org.apache.commons.jexl2.parser.ASTForeachStatement;
+import org.apache.commons.jexl2.parser.ASTIdentifier;
+import org.apache.commons.jexl2.parser.ASTIfStatement;
+import org.apache.commons.jexl2.parser.ASTIntegerLiteral;
+import org.apache.commons.jexl2.parser.ASTMapEntry;
+import org.apache.commons.jexl2.parser.ASTMapLiteral;
+import org.apache.commons.jexl2.parser.ASTMethodNode;
+import org.apache.commons.jexl2.parser.ASTModNode;
+import org.apache.commons.jexl2.parser.ASTMulNode;
+import org.apache.commons.jexl2.parser.ASTNullLiteral;
+import org.apache.commons.jexl2.parser.ASTNumberLiteral;
+import org.apache.commons.jexl2.parser.ASTReturnStatement;
+import org.apache.commons.jexl2.parser.ASTSizeFunction;
+import org.apache.commons.jexl2.parser.ASTSizeMethod;
+import org.apache.commons.jexl2.parser.ASTStringLiteral;
+import org.apache.commons.jexl2.parser.ASTTernaryNode;
+import org.apache.commons.jexl2.parser.ASTTrueNode;
+import org.apache.commons.jexl2.parser.ASTUnaryMinusNode;
+import org.apache.commons.jexl2.parser.ASTWhileStatement;
+import org.apache.commons.jexl2.parser.SimpleNode;
+
+public class ShortCircuitBaseVisitor extends BaseVisitor {
+    
+    @Override
+    public Object visit(SimpleNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTBlock node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTAmbiguous node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTIfStatement node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTWhileStatement node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTForeachStatement node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTTernaryNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTBitwiseOrNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTBitwiseXorNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTBitwiseAndNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTAdditiveNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTAdditiveOperator node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTMulNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTDivNode node, Object data) {
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTModNode node, Object data) {
+        return data;
+    }
```

Comment:
A visitor that needs to visit queries like the ones found in FunctionalSetTest.testFunctionsAsArguments() will not visit the whole tree. I think in most cases that is fine but if someone is doing something fancy with their query we could run into problems. Perhaps note that above in the class level documentation.

Reply:
Noted and will add a note in the docs